pkg/csi/service/wcpguest/controller.go

```diff
@@ -154,6 +179,19 @@ func (c *controller) Init(config *cnsconfig.Config) error {
 		log.Errorf("failed to watch on path: %q. err=%v", cnsconfig.DefaultpvCSIProviderPath, err)
 		return err
 	}
+	// Go module to keep the metrics http server running all the time.
+	go func() {
+		csiInfo.WithLabelValues(version).Set(1)
+		for {
+			log.Info("Starting the http server to expose Prometheus metrics..")
+			http.Handle("/metrics", promhttp.Handler())
+			err = http.ListenAndServe(":2112", nil)
```

Comment:
`2112` : Shouldn't this be configurable ?

Reply:
No, we need a fixed port so that we can expose the http server via a k8s Service. This is required for Prometheus to pull the metrices from the controller container.